packages/block-library/src/file/index.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * Server-side rendering of the `core/file` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * When the `core/file` block is rendering, check if we need to enqueue the `'wp-block-library-file` script.
+ *
+ * @param array $attributes The block attributes.
+ * @param array $content    The block content.
+ *
+ * @return string Returns the block content.
+ */
+function render_block_core_file( $attributes, $content ) {
+	if ( ! empty( $attributes['displayPreview'] ) ) {
```

Comment:
Nice. I don't know if it's something that could be abstracted as part of `register_block_type` call one day, but I like how this checks optimizes for the frontend 👍🏻

Reply:
It seems like it'd be a giant rabbit hole to abstract it, but I'd be interested to see an exploration. It'd be easy enough to say "load `frontend.js` if any of these attributes are set", but it'd get super complex for any use cases beyond that.

Reply:
Right, it can quickly get out of control. It's probably fine to leave it as is here and extend API of `register_block_type` to allow providing the frontend script that enqueues only when the block is rendered on the page.